### PR TITLE
clean up type nits on torch/jit/_ir_utils.py

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #143371

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### CODE REVIEW

**Review by aorenste — DISMISSED** [review]
Er - Sorry - this is py3.10 syntax. The union syntax needs to use `Union` instead of `|` since we still have to support 3.9...

File: torch/jit/_ir_utils.py [hunk 1]
```diff
@@ -17,7 +18,12 @@ def __enter__(self) -> None:
         self.prev_insert_point = self.g.insertPoint()
         self.g.setInsertPoint(self.insert_point)
 
-    def __exit__(self, *args: object) -> None:
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> None:
```
Comment (aorenste) [inline torch/jit/_ir_utils.py:26]:
```suggestion
    def __exit__(
        self,
        exc_type: Optional[Type[BaseException]],
        exc_val: Optional[BaseException],
        exc_tb: Optional[TracebackType],
    ) -> None:
```

**Comment by Skylion007** [comment]
I'm surprised this worked. The reason we didn't apply this codebase wide is that it caused issues with torch jit

**Review by laithsakka — APPROVED** [review]

**Pull request closed without merging**